SpatialGDK/Source/SpatialGDK/Private/Tests/RPCServiceTest.cpp

```diff
@@ -496,3 +508,128 @@ RPC_SERVICE_TEST(GIVEN_receiving_an_rpc_WHEN_return_false_from_extract_callback_
 	TestTrue("Returning false in extraction callback correctly stopped processing RPCs", bTestPassed);
 	return true;
 }
+
+DEFINE_LATENT_AUTOMATION_COMMAND_ONE_PARAMETER(FWaitForWorld, TSharedPtr<TestData>, Data);
+bool FWaitForWorld::Update()
+{
+	UWorld* World = nullptr;
+	const TIndirectArray<FWorldContext>& WorldContexts = GEngine->GetWorldContexts();
+	for (const FWorldContext& Context : WorldContexts)
+	{
+		if ((Context.WorldType == EWorldType::PIE || Context.WorldType == EWorldType::Game)
+			&& (Context.World() != nullptr))
+		{
+			World = Context.World();
+			break;
+		}
+	}
+
+	if (World != nullptr && World->AreActorsInitialized())
+	{
+		AGameStateBase* GameState = World->GetGameState();
+		if (GameState != nullptr && GameState->HasMatchStarted())
+		{
+			Data->TestWorld = World;
+			return true;
+		}
+	}
+
+	return false;
+}
+
+DEFINE_LATENT_AUTOMATION_COMMAND_ONE_PARAMETER(FSpawnActor, TSharedPtr<TestData>, Data);
+bool FSpawnActor::Update()
+{
+	FActorSpawnParameters SpawnParams;
+	SpawnParams.bNoFail = true;
+	SpawnParams.SpawnCollisionHandlingOverride = ESpawnActorCollisionHandlingMethod::AlwaysSpawn;
+
+	AActor* Actor = Data->TestWorld->SpawnActor<ATestActor>(SpawnParams);
+	Data->Actor = Actor;
+
+	return true;
+}
+
+DEFINE_LATENT_AUTOMATION_COMMAND_ONE_PARAMETER(FWaitForActor, TSharedPtr<TestData>, Data);
+bool FWaitForActor::Update()
+{
+	AActor* Actor = Data->Actor;
+	return (IsValid(Actor) && Actor->IsActorInitialized() && Actor->HasActorBegunPlay());
+}
+
+DEFINE_LATENT_AUTOMATION_COMMAND_TWO_PARAMETER(FDropRPCQueueTest, FAutomationTestBase*, Test, TSharedPtr<TestData>, Data);
+bool FDropRPCQueueTest::Update()
+{
+	if (!ensure(Test != nullptr))
+	{
+		return true;
+	}
+
+	if (!ensure(Data != nullptr) ||
+		!ensure(Data->Actor != nullptr))
+	{
+		Test->TestTrue("Returning false in extraction callback correctly stopped processing RPCs", false);
+	}
```

Comment:
I'm kinda confused by these / not seen this pattern before. Can we use `check(...)` instead or will that have different behaviour?

Reply:
Checks are fine but then we would crash after. As you can continue after checks I use ensures which will return whether the check was successful. This allows you to return from code that will crash.